File: src/main/java/com/order/ecommerce/model/Payment.java
package com.order.ecommerce.model;

import com.order.ecommerce.constants.PaymentMode;
import lombok.*;
import org.hibernate.annotations.Type;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.UUID;
import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(
        name = "ecommerce_payment"
)
public class Payment implements Serializable {
    @Id
    @Column(
            name = "payment_id",
            nullable = false,
            unique = true
    )
    @Type(type = "pg-uuid")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID paymentId;
    @Column(
            name = "amount",
            nullable = false
    )
    private double amount;
    @Column(
            name = "payment_mode",
            nullable = false
    )
    @Enumerated(EnumType.STRING)
    private PaymentMode paymentMode;
    @Column(
            name = "confirmation_number",
            nullable = false
    )
    private String confirmationNumber;
    @Column(
            name = "payment_status",
            nullable = false
    )
    private String paymentStatus;
    @Column(
            name = "createdAt",
            nullable = false
    )
    private LocalDate createdAt;

    @PrePersist
    private void prePersistFunction() {
        if (createdAt == null) {
            createdAt = LocalDate.now();
        }

        if (confirmationNumber == null) {
            confirmationNumber = UUID.randomUUID().toString();
        }
    }
}

File: src/main/kotlin/com/order/ecommerce/model/Payment.kt
package com.order.ecommerce.model

import com.order.ecommerce.enum.PaymentStatus
import java.io.Serializable
import java.time.LocalDate
import java.util.UUID
import javax.persistence.*

@Entity
@Table(name = "ecommerce_payment")
class Payment(

    @Id
    @Column(name = "payment_id", nullable = false, unique = true)
    private var paymentId: String,

    @Column(name = "amount", nullable = false)
    private var amount: Double,

    @Column(name = "payment_mode", nullable = false)
    private var paymentMode: String,

    @Column(name = "confirmation_number", nullable = false)
    private var confirmationNumber: String,

    @Column(name = "payment_status", nullable = false)
    private var paymentStatus: String,

    @Column(name = "createdAt", nullable = false)
    private var createdAt: LocalDate,

    @OneToOne(fetch = FetchType.LAZY, mappedBy = "payment")
    private var order: Order?

) : Serializable {

    constructor(amount: Double, paymentMode: String) : this(
        UUID.randomUUID().toString(),
        amount,
        paymentMode,
        UUID.randomUUID().toString(),
        PaymentStatus.PROCESSING.name,
        LocalDate.now(),
        null
    )
}
